src/client/table_client.rs

```diff
@@ -446,81 +461,73 @@ impl ObTableClientInner {
             }
             // Table not found, try to refresh it and retry get it again.
-            warn!("ObTableClientInner::get_tables can not get ob table by address {:?} so that will sync refresh metadata.",
+            warn!("ObTableClientInner::get_tables can not get ob table by address {:?} so that will refresh metadata.",
                   replica_location.addr());
-            self.sync_refresh_metadata()?;
-            let table_entry = self.get_or_refresh_table_entry(table_name, true)?;
-            match table_entry.get_partition_location_with_part_id(part_id) {
-                Some(location) => match location.leader() {
-                    Some(leader) => {
-                        //Found leader of replication ,try to get table from table roster
-                        if let Some(ob_table) = self.table_roster.rl().get(leader.addr()) {
-                            result.push((part_id, ob_table.clone()));
-                        } else {
-                            return Err(CommonErr(
-                                CommonErrCode::NotFound,
-                                format!("ObTable to {leader:?} not found in table_roster."),
-                            ));
-                        }
-                    }
-                    None => {
-                        //Leader not found
-                        return Err(CommonErr(
-                            CommonErrCode::NotFound,
-                            format!("Leader not found part_id={part_id} for table {table_entry:?}"),
-                        ));
-                    }
-                },
-                None => {
-                    //Replica not found.
-                    return Err(CommonErr(
-                        CommonErrCode::NotFound,
-                        format!(
-                            "Replica not found part_id={} for table {:?}",
-                            part_id,
-                            table_entry.to_owned(),
-                        ),
-                    ));
+
+            match self.refresh_sender.try_send(table_name.to_owned()) {
+                Ok(_) => {
+                    warn!("ObTableClientInner::get_tables: Need Refresh / try to refresh schema actively succeed, table_name:{table_name}");
+                }
+                Err(error) => {
+                    warn!("ObTableClientInner::get_tables: Need Refresh / try to refresh schema actively failed, maybe other thread has sent, table_name:{table_name}, error:{error}");
                 }
             }
+
+            let error_message = format!(
+                "get_tables can not get table by address {:?}, already try to refresh schema",
+                replica_location.addr()
+            );
+            return Err(CommonErr(CommonErrCode::NotFound, error_message));
         }
 
         Ok(result)
     }
 
-    fn fill_partition_location_with_part_id(
+    /*
+     * fill_partition_location_with_phy_id will return gt part_id in PartInfo
+     */
```

Comment:
Maybe we should follow the style of rust doc comments:
```suggestion
    /// fill_partition_location_with_phy_id will return gt part_id in PartInfo
```

BTW, the comment is confusing, shall we make it more readable?

Reply:
Fixed